flux-middle/src/rustc/lowering.rs

```diff
@@ -622,7 +622,13 @@ pub fn lower_ty<'tcx>(tcx: TyCtxt<'tcx>, ty: rustc_ty::Ty<'tcx>) -> Result<Ty, U
             let tys = List::from_vec(tys.iter().map(|ty| lower_ty(tcx, ty)).try_collect()?);
             Ok(Ty::mk_tuple(tys))
         }
-        rustc_ty::Array(ty, _) => Ok(Ty::mk_array(lower_ty(tcx, *ty)?, Const)),
+        rustc_ty::Array(ty, len) => {
+            let param_env = ParamEnv::empty().with_reveal_all_normalized(tcx);
+            let val = len.try_eval_usize(tcx, param_env).unwrap_or_else(|| {
```

Comment:
Ok, so this will fail if the length is a const generic

```rust
fn foo<const n: usize>(x: [i32; n]) { }
```
I think in general, instead of evaluating the length we should normalize the type with the appropriate `ParamEnv`. I'm not entirely sure, but I believe normalizing the type will evaluate the expression when possible. The correct `ParamEnv` will be either a generic one when checking the body of a function or an instantiated one when checking a function call. This is also related to supporting type projections. We can discuss this offline.

Reply:
Yes this is what I was referring to as the `ParamEnv` hack ...

Reply:
(So accept this for now and defer to later?) Let me add a TODO and link to this.

Reply:
Yes, that sounds good, let's merge it as is for now.